tests/test_litapi.py

```diff
@@ -79,3 +95,90 @@ def test_batch_unbatch_stream():
         count += 1
 
     assert count == 4, "Should have 4 responses"
+
+
+def test_decode_request():
+    request = {"input": 4.0}
+    api = ls.examples.SimpleLitAPI()
+    assert api.decode_request(request) == 4.0, "Decode request should return the input 4.0"
+
+
+def test_decode_request_with_openai_spec():
+    api = ls.examples.TestAPI()
+    api._sanitize(max_batch_size=1, spec=ls.OpenAISpec())
+    request = ChatCompletionRequest(messages=[{"role": "system", "content": "Hello"}])
+    decoded_request = api.decode_request(request)
+    assert decoded_request[0]["content"] == "Hello", "Decode request should return the input message"
+
+
+def test_decode_request_with_openai_spec_wrong_request():
+    api = ls.examples.TestAPI()
+    api._sanitize(max_batch_size=1, spec=ls.OpenAISpec())
+    with pytest.raises(AttributeError, match="object has no attribute 'messages'"):
```

Comment:
let's extend the failure modes here

Reply:
default decode_request is Pydantic typed so it handles anything other than a ChatCompletionRequest object with the same AttributeError.